app/lib/features/events/providers/event_providers.dart

```diff
@@ -131,21 +131,20 @@ enum EventFilters {
   past,
 }
 
-final eventFilterProvider =
-    StateProvider.autoDispose<EventFilters>((ref) => EventFilters.all);
-
 //SEARCH EVENTS
-typedef EventListSearchParams = ({String? spaceId, String searchText});
+typedef EventListSearchParams = ({
+  String? spaceId,
+  String searchText,
+  EventFilters eventFilter
+});
 
 final eventListSearchFilterProvider = FutureProvider.autoDispose
     .family<List<ffi.CalendarEvent>, EventListSearchParams>(
         (ref, params) async {
   //Declare filtered event list
   List<ffi.CalendarEvent> filteredEventList = [];
 
-  //Filter events based on the selection
-  EventFilters eventFilter = ref.watch(eventFilterProvider);
-  switch (eventFilter) {
+  switch (params.eventFilter) {
     case EventFilters.bookmarked:
```

Comment:
Why did you remove the provider (the proper pattern) and replace it with _several_ input parameters? The way local state is meant to be used is exactly the pattern that was existing before: a provider with a state, that is being updated and then all that is using that provider will update from the bottom up. Family is not meant to pass in local state like a search text or other volatile configuration.

Reply:
This is partly my bad, I didn't notice before that we did this change for the search text, but this going the wrong direction now.